### ISSUE TITLE
The sourcekit branch will not build.

### ISSUE THREAD

**Issue by person666** [RLovelett/swift-aur#14]
It spits out this. I do not have a separate  blocksruntime. Is this caused by that?

```
swift-aur/src/swift/tools/SourceKit/lib/Support/Concurrency-libdispatch.cpp:20:10: fatal error: 'Block.h' file not found
#include <Block.h>
         ^
1 error generated.
[22/47] Building CXX object tools/SourceKit/lib/SwiftLang/CMakeFiles/SourceKitSwiftLang.dir/SwiftEditorInterfaceGen.cpp.o
ninja: build stopped: subcommand failed.
utils/build-script: fatal error: command terminated with a non-zero exit status 1, aborting
```

**Comment by RLovelett** [comment]
Yeah the [sourcekit](https://github.com/RLovelett/swift-aur/tree/sourcekit) branch is still a work-in-progress.

You need to get a blocks runtime installed in order to build it. I don't really have "documentation" on how to do this. Basically find the source for the runtime (e.g., [mackyle/blocksruntime](https://github.com/mackyle/blocksruntime)) follow the instructions to build. Work-around any issues you run into compiling it. Then install it to your system includes directory. Then run this packager.

Bottom line: that branch is experimental and if you are building that branch you are expected to run into problems.

**Comment by person666** [comment]
did you ever run into this error?

```
/usr/bin/ld: /usr/local/lib/libBlocksRuntime.a(runtime.o): relocation R_X86_64_32S against symbol `_NSConcreteAutoBlock' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: final link failed: Nonrepresentable section on output
clang-3.8: error: linker command failed with exit code 1 (use -v to see invocation)
ninja: build stopped: subcommand failed.
utils/build-script: fatal error: command terminated with a non-zero exit status 1, aborting
```

I tried recompiling everything after `export CXXFLAGS="$CXXFLAGS -fPIC"` but the error persists.

**Comment by RLovelett** [comment]
I'm pretty sure this is what I used:

```
CC=clang CFLAGS=-fPIC ./buildlib
CC=clang CFLAGS=-fPIC ./checktests
sudo ./installlib
```

I _think_ I also had to copy `Block_private.h` because something uses a private header. Though do not hold me to that. I'm not 100% sure anymore.

**Comment by person666** [comment]
this worked. thanks

**Comment by person666** [comment]
I know this is not your repo, but did you ever run into [this error with SourceKitten](https://github.com/jpsim/SourceKitten/issues/290)?

**Comment by RLovelett** [comment]
Yes I have. Though I have not had time to track down the source. Not sure if it is due to some compiler flag/setting in my build, a bug in Foundation, or a bug in SourceKitten.

**Comment by person666** [comment]
[it](https://github.com/jpsim/SourceKitten/issues/290) is not a problem with your build exclusively at least. [The swift build in the aur](https://aur.archlinux.org/packages/swift/) also produces the same bug. I will test a build in Ubuntu and get back to you.

**Comment by person666** [comment]
SourceKitten seems to work ok on Ubuntu, building swift with what amounts to the same flags on Ubuntu works. I find it unlikely it is a problem with your build flags
